pkg/xds/envoy/clusters/health_check_configurer.go

```diff
@@ -20,19 +24,99 @@ type healthCheckConfigurer struct {
 	healthCheck *mesh_core.HealthCheckResource
 }
 
+func mapUInt32ToInt64Range(value uint32) *envoy_type.Int64Range {
+	return &envoy_type.Int64Range{
+		Start: int64(value),
+		End:   int64(value),
+	}
+}
+
+func mapHttpHeaders(
+	headers []*v1alpha1.HealthCheck_Conf_Http_HeaderValueOption,
+) []*envoy_core.HeaderValueOption {
+	var envoyHeaders []*envoy_core.HeaderValueOption
+	for _, header := range headers {
+		envoyHeaders = append(envoyHeaders, &envoy_core.HeaderValueOption{
+			Header: &envoy_core.HeaderValue{
+				Key:   header.Header.Key.Value,
+				Value: header.Header.Value.Value,
+			},
+			Append: header.Append,
+		})
+	}
+	return envoyHeaders
+}
+
 func (e *healthCheckConfigurer) Configure(cluster *envoy_api.Cluster) error {
 	if e.healthCheck == nil || e.healthCheck.Spec.Conf == nil {
 		return nil
 	}
 	activeChecks := e.healthCheck.Spec.Conf
-	cluster.HealthChecks = append(cluster.HealthChecks, &envoy_core.HealthCheck{
+	healthCheck := envoy_core.HealthCheck{
 		HealthChecker: &envoy_core.HealthCheck_TcpHealthCheck_{
 			TcpHealthCheck: &envoy_core.HealthCheck_TcpHealthCheck{},
 		},
```

Comment:
remove 29-31 since we might be putting HTTP Health Check

Reply:
That's true, but according to [the documentation](https://www.envoyproxy.io/docs/envoy/v1.16.0/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck) precisely one of the properties (`http_health_check`, `tcp_health_check`, `grpc_health_check`, `custom_health_check`) has to be set, so as we are defaulting to the `tcp` health checks this is the default value as you are able to not provide specific `http` or `tcp`config blocks and without this lines it generate incomplete configuration (it will also break the tests)

Reply:
On the other hand, we validate that the HealthCheck has an HTTP or TCP section.

but ok, technically you are right that it's more safe